nautobot/extras/forms/forms.py

```diff
@@ -982,9 +986,34 @@ class Meta:
             "has_sensitive_variables",
             "job_queues_override",
             "job_queues",
+            "default_job_queue_override",
+            "default_job_queue",
             "tags",
         ]
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        job_model = self.instance
+        job_queue_queryset = JobQueue.objects.filter(jobs=job_model)
+        job_queue_params = {"jobs": [job_model.pk]}
+        if job_model.default_job_queue:
+            default_job_queue = job_model.default_job_queue.pk
+        else:
+            # Populate the field with settings.CELERY_TASK_DEFAULT_QUEUE JobQueue if it exists
+            try:
+                default_job_queue = JobQueue.objects.get(name=settings.CELERY_TASK_DEFAULT_QUEUE)
+                default_job_queue = default_job_queue.pk
+            except JobQueue.DoesNotExist:
+                pass
+        self.fields["default_job_queue"] = DynamicModelChoiceField(
+            queryset=job_queue_queryset,
+            query_params=job_queue_params,
+            required=False,
+            help_text="The default job queue to route this job to",
+            label="Default Job queue",
+        )
+        self.fields["default_job_queue"].initial = default_job_queue
```

Comment:
This feels awry to me - since you can edit the `job_queues` and the `default_job_queue` at the same time, it doesn't make sense for the `default_job_queue` to constrain the queryset or query_params based on the preexisting job-queue assignments?

Hmm... this is similar actually to the recent discussion around setting a `VDC.primary_ip4` while also changing the `VDC.interfaces`. What if we changed things around so that the selected `default_job_queue` gets implicitly/automatically included into `job_queues` even if it wasn't explicitly specified as such by the user?